src/SonarQube.TeamBuild.PostProcessor/MSBuildPostProcessor.cs

```diff
@@ -164,6 +170,26 @@ private bool CheckEnvironmentConsistency(AnalysisConfig config, ITeamBuildSettin
             return true;
         }
 
+        /// <summary>
+        /// Credentials must be passed to both begin and end step (or not passed at all). If the credentials are passed to only
+        /// one of the steps the analysis will fail so let's fail-fast with an explicit message.
+        /// </summary>
+        private bool CheckCredentialsInCommandLineArgs(AnalysisConfig config, IAnalysisPropertyProvider provider)
+        {
+            var hasCredentialsInBegin = config.AdditionalConfig?.Any(c => ConfigSetting.SettingKeyComparer.Equals(c.Id,
+                ConfigSettingsExtensions.IsUsingCommandLineCredentialsKey)) ?? false;
+
+            var hasCredentialsInEnd = provider.TryGetProperty(SonarProperties.SonarUserName, out var _);
```

Comment:
I don't care about the property hence the `out var _`. I don't know if you are aware of this pattern for unused variables.

Reply:
Minor: it might be neater to add a "HasProperty" extension method to _AnalysisPropertyProviderExtensions_.